test/e2e/specs/editor/blocks/cover.spec.js

```diff
@@ -0,0 +1,267 @@
+			name: 'Block: Cover',
+		} );
+
+		const filename = await imageBlockUtils.upload(
+			coverBlock.getByTestId( 'form-file-upload-input' )
+		);
+		const fileBasename = path.basename( filename );
+
+		// Wait for the img's src attribute to be prefixed with http.
+		// Otherwise, the URL for the img src attribute starts is a placeholder
+		// beginning with `blob`.
+		await expect( async () => {
+			const src = await coverBlock.locator( 'img' ).getAttribute( 'src' );
+			expect( src.startsWith( 'http' ) ).toBe( true );
+		} ).toPass();
+
+		const backgroundImageURL = await coverBlock
+			.locator( 'img' )
+			.getAttribute( 'src' );
+
+		expect( backgroundImageURL ).toContain( fileBasename );
+	} );
+
+	test( 'dims background image down by 50% by default', async ( {
+		page,
+		editor,
+		imageBlockUtils,
+	} ) => {
+		await editor.insertBlock( { name: 'core/cover' } );
+		const coverBlock = page.getByRole( 'document', {
+			name: 'Block: Cover',
+		} );
+
+		await imageBlockUtils.upload(
+			coverBlock.getByTestId( 'form-file-upload-input' )
+		);
+
+		const [ backgroundDimColor, backgroundDimOpacity ] =
+			await getBackgroundColorAndOpacity(
+				coverBlock.locator( 'span[aria-hidden="true"]' )
+			);
+		expect( backgroundDimColor ).toBe( 'rgb(0, 0, 0)' );
+		expect( backgroundDimOpacity ).toBe( '0.5' );
+	} );
+
+	test( 'can have the title edited', async ( { page, editor } ) => {
+		const titleText = 'foo';
+
+		await editor.insertBlock( { name: 'core/cover' } );
+		const coverBlock = page.getByRole( 'document', {
+			name: 'Block: Cover',
+		} );
+
+		// Choose a color swatch to transform the placeholder block into
+		// a functioning block.
+		await coverBlock
+			.getByRole( 'button', {
+				name: 'Color: Black',
+			} )
+			.click();
+
+		// Activate the paragraph block inside the Cover block.
+		// The name of the block differs depending on whether text has been entered or not.
+		const coverBlockParagraph = coverBlock.getByRole( 'document', {
+			name: /Paragraph block|Empty block; start writing or type forward slash to choose a block/,
+		} );
+		await expect( coverBlockParagraph ).toBeEditable();
+
+		await coverBlockParagraph.fill( titleText );
+
+		await expect( coverBlockParagraph ).toContainText( titleText );
+	} );
+
+	test( 'can be resized using drag & drop', async ( { page, editor } ) => {
+		await editor.insertBlock( { name: 'core/cover' } );
+		const coverBlock = page.getByRole( 'document', {
+			name: 'Block: Cover',
+		} );
+		await coverBlock
+			.getByRole( 'button', {
+				name: 'Color: Black',
+			} )
+			.click();
+
+		// Open the document sidebar.
+		await editor.openDocumentSettingsSidebar();
+
+		// Open the block list viewer from the toolbar.
+		await page
+			.getByRole( 'toolbar', { name: 'Document tools' } )
+			.getByRole( 'button', { name: 'Document Overview' } )
+			.click();
+
+		// Select the Cover block from the docuemnt overview.
+		await page.getByRole( 'link', { name: 'Cover' } ).click();
```

Comment:
We can target the list view using `getByRole('region', { name: 'Document Overview' }` or `getByRole('treegrid', { name: 'Block navigation structure' })`. So it's a bit clearer which link we are clicking on :).

Reply:
Got it. I have changed all the "raw" selectors to be anchored off a parent element. 

When there are multiple parent elements (like in this case) I chose the one that I think makes sense.

Addressed in b6e5c294eb63aa6f79bc364254f622f75aac533d.